src/borg/remote.py

```diff
@@ -1220,8 +1209,7 @@ def unpack(data):
             csize, checksum = cache_struct.unpack(data[:cache_struct.size])
             compressed = data[cache_struct.size:]
             if checksum != xxh64(compressed):
-                logger.warning('Repository metadata cache: detected corrupted data in cache!')
-                raise RepositoryCache.InvalidateCacheEntry
+                raise IntegrityError('detected corrupted data in metadata cache')
```

Comment:
hmm, just as a vague idea: could we get into similar troubles by this as we had with OSError?

i mean: we shouldn't kill a repo chunk after confusing a repo chunk IE with a cache IE... (this somehow fits into the scope of the ticket "review IE, especially when they kill").